ci/jenkinsfiles/release.groovy

```diff
@@ -0,0 +1,335 @@
+          }
+        }
+      }
+    }
+    stage('Linting') {
+      steps {
+        container('maven') {
+          script {
+            pipelineLib.lint()
+          }
+        }
+      }
+    }
+    stage('Run Unit Tests') {
+      steps {
+        script {
+          def stages = [:]
+          stages['backend'] = pipelineLib.runBackEndUnitTests()
+          stages['frontend'] = pipelineLib.runFrontEndUnitTests()
+          parallel stages
+        }
+      }
+    }
+    stage('Publish') {
+      when {
+        allOf {
+          not {
+            branch 'PR-*'
+          }
+          not {
+            environment name: 'DRY_RUN', value: 'true'
+          }
+        }
+      }
+      environment {
+        MESSAGE = "Release ${VERSION}"
+        TAG = "v${VERSION}"
+      }
+      stages {
+        stage('Git Commit and Tag') {
+          steps {
+            container('maven') {
+              script {
+                pipelineLib.gitCommit("${MESSAGE}", '-a')
+                pipelineLib.gitTag("${TAG}", "${MESSAGE}")
+              }
+            }
+          }
+        }
+        stage('Publish Coldstorage Package') {
+          steps {
+            container('maven') {
+              script {
+                echo """
+                  -------------------------------------------------
+                  Upload Coldstorage Package ${VERSION} to ${CONNECT_PROD_URL}
+                  -------------------------------------------------
+                """
+                if (DRY_RUN_RELEASE == 'false') {
+                  pipelineLib.uploadPackage("${VERSION}", 'connect-preprod', "${CONNECT_PROD_URL}")
+                }
+              }
+            }
+          }
+          post {
+            always {
+              archiveArtifacts (
+                allowEmptyArchive: true,
+                artifacts: 'nuxeo-coldstorage-package/target/nuxeo-coldstorage-package-*.zip'
+              )
+            }
+          }
+        }
+        stage('Git Push') {
+          steps {
+            container('maven') {
+              echo """
+                --------------------------
+                Git push ${TAG}
+                --------------------------
+              """
+              script {
+                if (DRY_RUN_RELEASE == 'false') {
+                  pipelineLib.gitPush("${TAG}")
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    stage('Release and Bump reference branch') {
+      when {
+        allOf {
+          not {
+            environment name: 'DRY_RUN', value: 'true'
+          }
+        }
+      }
+      steps {
+        container('maven') {
+          script {
+            //cleanup files updated by other stages
+            sh 'git reset --hard'
+            // increment minor version
+            String nextVersion =
+              sh(returnStdout: true, script: "perl -pe 's/\\b(\\d+)(?=\\D*\$)/\$1+1/e' <<< ${CURRENT_VERSION}").trim()
```

Comment:
Perhapas I am missing something, but can we use the `nxp`, `semver` tools that we used on `rc` publication

Reply:
we can try.

Reply:
I think `semver` will complain about the version format

```
Arguments:
  <version>  A version must match the following regular expression:
             "^[vV]?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(\-(0|[1-9][0-9]*|[0-9]*[A-Za-z-][0-9A-Za-z-]*)(\.(0|[1-9][0-9]*|[0-9]*[A-Za-z-][0-9A-Za-z-]*))*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$"
             In English:
             -- The version must match X.Y.Z[-PRERELEASE][+BUILD]
```

I also remember that we have the same problem with nxp.

Reply:
See my comment https://github.com/nuxeo/nuxeo-coldstorage/pull/12#discussion_r543405838